calvin/calvin.py

```diff
@@ -1,151 +1,146 @@
-          stop=stop,
-          temperature=data[prompt][0]['temperature'],
-          logprobs=0,
-      )
-      choices += completion.choices
-
-  def score(a):
-      return sum(a.logprobs.token_logprobs) / len(a.logprobs.token_logprobs)
-
-  choices = sorted(choices, key=lambda a: -score(a))
-  for i, choice in enumerate(choices):
-      print(f"======")
-      print(choice.text.strip())
-
-  return 0
-
-if __name__ == "__main__":
-    prompt = input('Enter prompt: ')
-    sys.exit(main(prompt))
\ No newline at end of file
+# Defile module-scoped variables
+engine = "davinci"
+headers = {}
+prompts = {}
+
+
+# Initialize module
+def initialize(_api_key, _engine):
+  global engine, headers, prompts
+
+  if _engine:
+    engine = _engine
+
+  headers = {
+    "Content-Type": "application/json",
+    "Authorization": f"Bearer {_api_key}"
+  }
+
+  prompts = json.load(open("./calvin/prompts.json"))
+
+#
+def get_globals():
+  global engine, headers, prompts
+
+  print({
+    engine: engine,
+    headers: headers,
+    prompts: prompts # Figure out why this wont serialize -_-
+  })
+
+# Change engine setting
+def set_engine(_engine):
+  global engine
+  engine = _engine
+
+# List Engines GET
+# Lists the currently available engines, and provides basic information about each option such as the owner and availability.
+def list_engines():
+  global headers
+
+  url = "https://api.openai.com/v1/engines"
+  response = requests.get(url, headers=headers)
+
+  if response.ok:
+    print(f"JSON: {response.json()}")
+    return response.json()
+  else:
+    return response.raise_for_status()
+
+# Retrieve Engine GET
+# Retrieves an engine instance, providing basic information about the engine such as the owner and availability.
+def retrieve_engine():
+  global headers
+
+  url = f"https://api.openai.com/v1/engines/{engine}"
+  response = requests.get(url, headers=headers)
+
+  if response.ok:
+    print(f"JSON: {response.json()}")
+    return response.json()
+  else:
+    return response.raise_for_status()
+
+# Create Completion POST
+# Returns new text as well as, if requested, the probabilities over each alternative token at each position.
+def create_completion(payload):
+  global engine, headers
+
+  url = f"https://api.openai.com/v1/engines/{engine}/completions"
+  response = requests.post(url, headers=headers, json=payload)
+
+  if response.ok:
+    print(f"JSON: {response.json()}")
+    return response.json()
+  else:
+    return response.raise_for_status()
+
+def complete_prompt(prompt):
+  payload = {
+    "prompt": prompt,
+    "max_tokens": 5,
+    "temperature": 1,
+    "top_p": 1,
+    "n": 1
+  }
+
+  create_completion(payload)
+
+def complete_predefined_prompt(prompt_key, index=0):
+  global prompts
+
+  if prompts[prompt_key] in prompts:
+    create_completion(prompts[prompt_key][index])
+
+  return "Unable to access predefined prompt."
+
+
+# Search POST
+# Perform a semantic search over a list of documents.
+def search(payload):
```

Comment:
Have we tested this? 
![image](https://user-images.githubusercontent.com/11607499/88253212-2876ab00-cc7f-11ea-8ae4-a1aa0fb68331.png)

Seems really powerful.